esp-hal/src/spi/master.rs

```diff
@@ -2172,9 +2172,8 @@ mod ehal1 {
                     SpiBus::write(self, &write[write_from..write_to])?;
                 }
 
-                SpiBus::flush(self)?;
-
                 if read_inc > 0 {
+                    SpiBus::flush(self)?;
```

Comment:
Gratis technicality: if we read less than we write, we are allowed to write the rest in the background.